src/BloomFilter.cpp

```diff
@@ -14,95 +14,108 @@
  * limitations under the License.
  */
 
+#include <climits>
+#include <cmath>
 #include <cstdio>
 #include <fstream>
-#include <cmath>
-#include <assert.h>
 #include "BloomFilter.hpp"
 
+static const size_t BITS_PER_BLOCK = 8;
+
 // Forward declarations
 
+static void checkArchitecture();
+
 static size_t calculateHashRounds(size_t size, size_t maxItems);
 
-static unsigned int djb2Hash(string text);
+static unsigned int djb2Hash(const string &text);
 
-static unsigned int sdbmHash(string text);
+static unsigned int sdbmHash(const string &text);
 
 static unsigned int doubleHash(unsigned int hash1, unsigned int hash2, unsigned int round);
 
-static void writeVectorToStream(vector<bool> &bloomVector, BinaryOutputStream &out);
-
-static BlockType pack(const vector<bool> &filter, size_t block, size_t bits);
+static vector<BlockType> readVectorFromFile(const string &path);
 
-static vector<bool> readVectorFromFile(const string &path);
-
-static vector<bool> readVectorFromStream(BinaryInputStream &in);
-
-static void unpackIntoVector(vector<bool> &bloomVector,
-                             size_t offset,
-                             size_t bitsInThisBlock,
-                             BinaryInputStream &in);
+static vector<BlockType> readVectorFromStream(BinaryInputStream &in);
 
 
 // Implementation
 
 BloomFilter::BloomFilter(size_t maxItems, double targetProbability) {
-    auto size = (size_t) ceil((maxItems * log(targetProbability)) / log(1.0 / (pow(2.0, log(2.0)))));
-    bloomVector = vector<bool>(size);
-    hashRounds = calculateHashRounds(size, maxItems);
+    checkArchitecture();
+    bits = (size_t) ceil((maxItems * log(targetProbability)) / log(1.0 / (pow(2.0, log(2.0)))));
+    auto blocks = (size_t) ceil(bits / (double) BITS_PER_BLOCK);
+    bloomVector = vector<BlockType>(blocks);
+    hashRounds = calculateHashRounds(bits, maxItems);
 }
 
-BloomFilter::BloomFilter(string importFilePath, size_t maxItems) {
+BloomFilter::BloomFilter(const string &importFilePath, size_t bits, size_t maxItems) {
+    checkArchitecture();
+    this->bits = bits;
```

Comment:
Optional: There is a nice syntactic sugar when initialising properties with constructor arguments:
`BloomFilter::BloomFilter(const string &importFilePath, size_t bits, size_t maxItems): bits(bits) {
...
`
No need to apply here :) just to let you know

Reply:
I like it. Done!